internal/rest/utils/params.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2021 Kaleido
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/hyperledger-labs/firefly-fabconnect/internal/errors"
+	"github.com/hyperledger-labs/firefly-fabconnect/internal/utils"
+	"github.com/julienschmidt/httprouter"
+)
+
+func ResolveParams(res http.ResponseWriter, req *http.Request, params httprouter.Params) (c map[string]interface{}, err error) {
+	// if query parameters are specified, they take precedence over body properties
+	channelParam := params.ByName("channel")
+	methodParam := params.ByName("func")
+	signer := GetFlyParam("signer", req, false)
+	blocknumber := GetFlyParam("blocknumber", req, false)
+
+	body, err := utils.ParseJSONPayload(req)
+	if err != nil {
+		errors.RestErrReply(res, req, err, 400)
+		return nil, err
+	}
+
+	// consolidate inidividual parameters with the body parameters
+	if channelParam != "" {
+		body["headers"].(map[string]string)["channel"] = channelParam
+	}
+	if methodParam != "" {
+		body["func"] = methodParam
+	}
+	if signer != "" {
+		body["headers"].(map[string]string)["signer"] = signer
+	}
+	if blocknumber != "" {
+		body["headers"].(map[string]string)["blocknumber"] = blocknumber
+	}
+
+	return body, nil
+}
+
+// getFlyParam standardizes how special 'fly' params are specified, in query params, or headers
+func GetFlyParam(name string, req *http.Request, isBool bool) string {
+	valStr := ""
+	vs := getQueryParamNoCase(utils.GetenvOrDefaultLowerCase("PREFIX_SHORT", "fly")+"-"+name, req)
```

Comment:
Should we let viper handle the environment variables / defaults? This is what we do in `firefly` core.

Reply:
this line isn't really consuming the configurations, it's figuring out the prefix that is only optionally set via an environment variable (background https://github.com/hyperledger-labs/firefly-ethconnect/pull/118#pullrequestreview-674348896)

Reply:
Right. FireFly and ethconnect handle these a bit differently. In FireFly, the long and short prefixes are actually managed by Viper. Which means they can be set in the config file, or they can be overridden by environment variables.
https://github.com/hyperledger-labs/firefly/blob/main/internal/blockchain/ethereum/config.go#L46-L49

I'm not fundamentally opposed to the way the current code is here, but it was just a suggestion for a potential improvement.